pg_checksums.c

```diff
@@ -286,17 +347,21 @@ scan_file(char *fn, int segmentno)
 				fprintf(stderr, _("%s: write failed: %s\n"), progname, strerror(errno));
 				exit(1);
 			}
+
+			if (show_progress)
+				report_progress(false);
 		}
 	}
 
 	close(f);
 }
 
-static void
-scan_directory(char *basedir, char *subdir)
+static int64
+scan_directory(char *basedir, char *subdir, bool sizeonly)
 {
-	char		path[MAXPGPATH];
-	DIR		   *dir;
+	int64          dirsize = 0;
+	char		   path[MAXPGPATH];
+	DIR		      *dir;
```

Comment:
dirsize is new, but the other two look like unrelated whitespace changes, did you check that those are legit?